lib/cors-anywhere.js

```diff
@@ -371,6 +372,13 @@ function getHandler(options, proxy) {
       return;
     }
 
+    var targetHost = location.host;
+    if (corsAnywhere.targetHostWhitelist.length && corsAnywhere.targetHostWhitelist.indexOf(targetHost) === -1) {
```

Comment:
This logic belongs to `proxyRequest`, otherwise a redirect can be abused to bypass this.

Alternatively, disable following of redirects.